src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,156 @@
+
+            try {
+                boolean b = affectedRows < 1;
+            } catch (DataProcessingException ex) {
+                throw new DataProcessingException("Expected to insert at least one row, "
+                        + "but inserted 0 rows.", ex);
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Dao
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                Long rowId = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book();
+                book.setId(rowId);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection", e);
+        }
+        return Optional.empty();
+    }
+
+    @Dao
+    @Override
+    public List<Book> findAll() {
+        Long id = 0L;
+        List<Book> bookList = new ArrayList<>();
+
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet;
+
+            do {
+                resultSet = preparedStatement.executeQuery();
+                Long rowId = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book();
+                book.setId(rowId);
+                book.setTitle(title);
+                book.setPrice(price);
+
+                bookList.add(book);
+                id++;
+            } while (resultSet.next());
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection", e);
+        }
+        return bookList;
+    }
+
+    @Dao
+    @Override
+    public Book update(Book book) {
+        String sql = "SELECT * FROM books WHERE title = ?";

```

Comment:
The SQL query in `update` is incorrect. It should be an `UPDATE` statement, not a `SELECT`. You need to update the book's fields in the database based on its ID.

Reply:
ok, done